pkg/controller/lifecycle/actuator.go

```diff
@@ -282,6 +310,55 @@ func (a *actuator) getShootAgentResources(defaultPeriod time.Duration, pingEnabl
 	return shootResources, nil
 }
 
+func buildAgentNetworkPolicy() client.Object {
+	tcp := corev1.ProtocolTCP
+	podGRPCPort := intstr.FromInt(common.PodNetPodGRPCPort)
+	podHttpPort := intstr.FromInt(common.PodNetPodHttpPort)
+	hostGRPCPort := intstr.FromInt(common.HostNetPodGRPCPort)
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gardener.cloud--allow-to-from-nwpd-agents",
+			Namespace: constants.NamespaceKubeSystem,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"networking.gardener.cloud/to-from-nwpd-agents": "allowed",
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &tcp,
+							Port:     &podGRPCPort,
+						},
+						{
+							Protocol: &tcp,
+							Port:     &hostGRPCPort,
+						},
+					},
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
```

Comment:
Can we add a `podSelector` here?

Reply:
for ingress a `podSelector` makes no sense, as these ports are also accessed by Prometheus and the `nwpdcli`.